db/migrate/20170712033920_add_language_to_users.rb

```diff
@@ -0,0 +1,5 @@
+class AddLanguageToUsers < ActiveRecord::Migration[5.0]
+  def change
+    add_column :users, :language, :string
```

Comment:
I think it is better if registered user can set their language preference,  not just detecting a browser preference.

In my case, I like to set my browser preference to Japanese, that is my mother tang, but sometimes want to see English version of site.  It is cumbersome to change browser setting before visiting site when see English version messages, as a site Trasnlator or just an interest.

Anonymous user can not set their language permanently, but registerd user can do it. It may be acceptable for all users.